src/main/scala/org/constellation/storage/SnapshotService.scala

```diff
@@ -273,36 +272,50 @@ class SnapshotService[F[_]: Concurrent](
       .map(_.hash)
       .map(hash => Snapshot(hash, hashesForNextSnapshot))
 
-  private[storage] def applySnapshot()(implicit C: ContextShift[F]): F[Unit] =
-    snapshot.get.flatMap { currentSnapshot =>
-      if (currentSnapshot == Snapshot.snapshotZero) {
-        Sync[F].unit
-      } else {
-        writeSnapshotToDisk(currentSnapshot) >>
-          applyAfterSnapshot(currentSnapshot) >>
-          dao.metrics.incrementMetricAsync(Metrics.snapshotCount)
+  private[storage] def applySnapshot()(implicit C: ContextShift[F]): EitherT[F, SnapshotError, Unit] = {
+    val write: Snapshot => F[Either[SnapshotError, Unit]] = (currentSnapshot: Snapshot) =>
+      writeSnapshotToDisk(currentSnapshot)
+        .flatTap(
```

Comment:
Why `applyAfterSnapshot`  is in `flatTap` which should be used for executing side effects?

Reply:
fixed